### ISSUE TITLE
using RCL partial views from outside the RCL

### ISSUE THREAD

**Issue by joeaudette** [aspnet/Razor#2391]
I left a comment in the documentation https://github.com/aspnet/Docs/issues/6918, but thought it would be better to ask about it here.

From the documentation:

> For content outside the RCL, the ASP.NET Core runtime does not search for partial files in the RCL.

To me that is a big limitation. I have a library of commonly used things, one example is Alerts, and the view for Alerts is naturally a partial view. Consumers of my library should be able to use that view without having to copy it local to their project.

I also have some other views that bundle commonly used scripts and css for things like datepickers and wysiwyg editors. The static resources are served as embedded resources from my RCL but the partials cannot be used outside my RCL as I want them to be.

Is there a way around that limitation such as an interface or view location expander I could implement to change that? 
What is the rationale for this limitation?

**Comment by joeaudette** [comment]
I can see that for the Identity UI library this same issue forced the need for inclusion of the _LoginPartial directly in the project template since it is consumed from the _Layout file it therefore can't be consumed from the Identity UI RCL and has to be scaffolded into the web app. 

Now multiply this problem for all library authors who wish to share common partial views but instead must somehow deliver the needed partials into the consuming app in order for their library to work. We don't really have the option to scaffold our partial views into the app.

**Comment by joeaudette** [comment]
Hmm, getting some unexpected success. I moved my AlertsPartial into another RCL in the same solution, it is invoked from the _Layout file that lives in the main web app and the AlertsPartial.cshtml does not exist in the the main web app yet it works and is apparently found in the RCL.

I wonder if this is because the RCL I moved it to also has an _Layout file, so the _Layout file in the main app is essentially overriding the one from the RCL and I guess that allows it to find the AlertsPartial in the RCL?

**Comment by joeaudette** [comment]
Closing this as it turned out to be false information in the docs, but the docs corroborated an experiment that did not work for me. After finding out the docs were incorrect I revisited my initial failed experiment and it worked this time though not sure what I did wrong the first time.

**Comment by joeaudette** [comment]
I think there is still some obscure bug here. As @Rick-Anderson noted, he documented that partial views could not be invoked from outside the RCL because he could not get it to work. I encountered the same thing initially and then the docs reinforced the idea that it was not supported. Later trying again after learning it should work I got it working.

But now I've encountered the problem again in a different solution/project. I have a viewcomponent in an RCL in my solution and the main web app has a project reference to the RCL project, but invoking the viewcomponent from the _Layout file in the main web app it fails to find the view in the RCL.
On the other hand I have other projects and solutions with the same scenario that do work and I have no idea why this one doesn't work.
I should note that I'm not using Razor Pages, I'm just using normal razor views with view models and mvc controllers and view components. In my RCL projects I have my views in the normal structure of Views folder Views/Shared/Components etc.

**Comment by joeaudette** [comment]
Ok, I figured out the issue. My first attempt I was updating an existing class library project that I previously used for embedded views to make it an RCL using pre-compiled views I uppdated the project Sdk to Microsoft.NET.Sdk.Razor and the only package reference I had was:

    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor" Version="2.1.*" />

So I noticed if I create a new RCL project it gets all of Mvc like this:

    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.*" />

and updating my existing project like that solved it. So closing this again.

On a separate note it is slightly confusing that to create a new RCL we have to first choose Asp.Net core Web "Application" then choose RCL from the dialog when RCL is not really an "Application". It is ok once you know where to find it, but would be easier to find if it were directly on the web project node in VS.
